storage/blockdevice/tests/TESTS/blockdevice/general_block_device/main.cpp

```diff
@@ -546,11 +546,18 @@ void test_contiguous_erase_write_read()
 
     bd_size_t contiguous_erase_size = stop_address - start_address;
     TEST_ASSERT(contiguous_erase_size > 0);
-    utest_printf("contiguous_erase_size=%d\n", contiguous_erase_size);
+    utest_printf("contiguous_erase_size=0x%" PRIx64 "\n", contiguous_erase_size);
 
     bd_size_t write_read_buf_size = program_size;
-    if (contiguous_erase_size / program_size > 8 && contiguous_erase_size % (program_size * 8) == 0) {
-        write_read_buf_size = program_size * 8;
+
+    // Reading/writing in larger chunks reduces the number of operations,
+    // helping to avoid test timeouts. Try 256-byte chunks if contiguous_erase_size
+    // (which should be a power of 2) is greater than that. If it's less than
+    // that, the test finishes quickly anyway...
+    if ((program_size < 256) && (256 % program_size == 0)
+        && (contiguous_erase_size >= 256) && (contiguous_erase_size % 256 == 0)) {
```

Comment:
Checks of `(256 % program_size == 0)` and `(contiguous_erase_size % 256 == 0)` shouldn't be necessary, because sector and program sizes are always powers of 2. We're just trying to be rigorous...